tmva/sofie/inc/TMVA/ROperator_Concat.hxx

```diff
@@ -112,10 +112,17 @@
             }
             std::stringstream out;
             out<<"\n//--------- Concat\n";
-            // special case for 0 axis that memory is contigous
-            if (fAxis == 0) {
-               for(auto &it : fInputs){
-                  out<<SP<<"tensor_" << fOutput<<".insert("<<"tensor_" << fOutput<<".end(),tensor_" << it<<".begin(),tensor_"<<it<<".end());\n";
+            // special case when memory is contigous
+            bool hasShapeOnes = true;
```

Comment:
I think here the loop should be on the tensor dimensions and not on the inputs. All inputs have by definition the same shape, no need to loop on them.
```
for(size_t i = 0; i<fAxis; ++i){
                   if(fInputShapes[0][i]!=1)
                   hasShapeOnes = false;
                   break;  // no need to continue the loop
  }

Reply:
yes, I agree. Made the change, thanks!